File: backend/TradeX.Core/Models/Exchange.cs
using TradeX.Core.Abstractions;
using TradeX.Core.Enums;
using TradeX.Core.Events;

namespace TradeX.Core.Models;

public enum ExchangeStatus
{
    Enabled,
    Disabled
}

public class Exchange : AggregateRoot
{
    // EF Core 无参构造函数
    public Exchange() { }

    /// <summary>工厂方法：创建交易所配置。</summary>
    public static Exchange Create(Guid createdBy, string name, ExchangeType type,
        string apiKeyEncrypted, string secretKeyEncrypted, string? passphraseEncrypted = null,
        Guid? traderId = null)
    {
        return new Exchange
        {
            CreatedBy = createdBy,
            Name = name,
            Type = type,
            ApiKeyEncrypted = apiKeyEncrypted,
            SecretKeyEncrypted = secretKeyEncrypted,
            PassphraseEncrypted = passphraseEncrypted,
            TraderId = traderId
        };
    }

    public Guid Id { get; init; } = Guid.NewGuid();
    public Guid? TraderId { get; init; }
    public string Name { get; set; } = string.Empty;
    public ExchangeType Type { get; set; }
    public string ApiKeyEncrypted { get; set; } = string.Empty;
    public string SecretKeyEncrypted { get; set; } = string.Empty;
    public string? PassphraseEncrypted { get; set; }
    public ExchangeStatus Status { get; set; } = ExchangeStatus.Enabled;
    public DateTime? LastTestedAt { get; set; }
    public string? TestResult { get; set; }
    public Guid CreatedBy { get; init; }
    public DateTime CreatedAt { get; init; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

    // ─────────────── 领域方法 ───────────────

    /// <summary>启用交易所。</summary>
    public void Enable()
    {
        var old = Status.ToString();
        Status = ExchangeStatus.Enabled;
        UpdatedAt = DateTime.UtcNow;
        AddDomainEvent(new ExchangeConnectionChangedEvent(Id, TraderId, old, Status.ToString()));
    }

    /// <summary>禁用交易所。</summary>
    public void Disable()
    {
        var old = Status.ToString();
        Status = ExchangeStatus.Disabled;
        UpdatedAt = DateTime.UtcNow;
        AddDomainEvent(new ExchangeConnectionChangedEvent(Id, TraderId, old, Status.ToString()));
    }

    /// <summary>记录连接测试结果。</summary>
    public void RecordTestResult(bool success, string? message = null)
    {
        LastTestedAt = DateTime.UtcNow;
        TestResult = message ?? (success ? "OK" : "Failed");
        UpdatedAt = DateTime.UtcNow;
    }
}

File: backend-go/internal/infra/ent/schema/exchange.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/field"
	"github.com/google/uuid"
)

// Exchange 交易所配置（含加密后的密钥）。对应 C# TradeX.Core.Models.Exchange。
//
// 列名严格对齐 C# EF 迁移建立的 exchanges 表（共享同一库）。索引（name 唯一、status）
// 由 C# 维护，此处不声明，避免 AutoMigrate 追加重复索引。
type Exchange struct {
	ent.Schema
}

func (Exchange) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).Default(uuid.New),
		field.String("name").Default(""),
		field.String("type").Default(""),
		field.Text("api_key_encrypted").Default(""),
		field.Text("secret_key_encrypted").Default(""),
		field.Text("passphrase_encrypted").Optional().Nillable(),
		field.String("status").Default("Enabled"),
		field.Time("last_tested_at").Optional().Nillable(),
		field.String("test_result").Optional().Nillable(),
		field.UUID("created_by", uuid.UUID{}).Default(uuid.New),
		field.Time("created_at").Default(time.Now).Immutable(),
		field.Time("updated_at").Default(time.Now).UpdateDefault(time.Now),
	}
}
